### ISSUE TITLE
max time feature

### ISSUE THREAD

**Issue by droberson** [ffuf/ffuf#85]
It would be nice to have an feature similar to curl's --max-time option to bail out if ffuf has been running for more than N seconds.

**Comment by joohoi** [comment]
Do you think it would be important to do so? You are currently able to infer the `--max-time` from the `--timeout` value * inputs length. Additionally, the main case for ffuf runs to take a long time are errors, and ffuf already has few cli flags to exit on spurious errors: `-sa`, `-se`, `-sf`.

**Comment by droberson** [comment]
Thank you very much for the workaround. I still think it would be nice to have this feature. If anything, it would be a lazy stopgap to make it so I don't have to do math problems before I scan a host.

This issue came up a few times when I was scanning lists of hosts using cheezy bash loops:

for i in $(cat hosts); do ffuf -w wordlist -u $i/FUZZ -o $i; done

One host right in the middle of the list was responding very slowly. it was not throwing errors; it was responding like it was being served from a 9600bps modem. 45 minutes later, i hit ^C  to hopefully just kill this ffuf process, but it killed my loop. This caused me to shout hateful profanities at my terminal. Thankfully no one was around to hear.

I edited my list, removing the slow host and the hosts I had already scanned, then IT HAPPENED AGAIN. This made me think that having a feature like curl's --max-time would be useful.
